src/platform_impl/linux/wayland/window/mod.rs

```diff
@@ -170,18 +170,19 @@ impl Window {
             )
             .map_err(|_| os_error!(OsError::WaylandMisc("failed to create window.")))?;
 
-        // Set CSD frame config
         #[cfg(feature = "sctk-adwaita")]
         {
-            let theme = platform_attributes.csd_theme.unwrap_or_else(|| {
-                let env = std::env::var(WAYLAND_CSD_THEME_ENV_VAR).unwrap_or_default();
-                match env.to_lowercase().as_str() {
-                    "dark" => Theme::Dark,
-                    _ => Theme::Light,
-                }
-            });
-
-            window.set_frame_config(theme.into());
+            // Set CSD frame config from theme if specified,
+            // otherwise use upstream automatic selection.
+            if let Some(theme) = platform_attributes.csd_theme.or_else(|| {
+                std::env::var(WAYLAND_CSD_THEME_ENV_VAR)
+                    .ok()?
+                    .as_str()
+                    .try_into()
+                    .ok()
+            }) {
+                window.set_frame_config(theme.into());
+            }
```

Comment:
You can shorten it like that, I think. I'm also not sure that you need `as_str` given that there's a `deref` impl.

```suggestion
            if let Ok(theme) = platform_attributes.csd_theme.ok_or_else(|| {
                std::env::var(WAYLAND_CSD_THEME_ENV_VAR).and_then(|var| var.as_str().try_into())
            }) {
                window.set_frame_config(theme.into());
            }
```

Reply:
Ok I've shortened the original commit's 7 extra lines to 30 extra lines now. Hopefully that's short enough now :laughing: